lib/builtin.t

```diff
@@ -67,6 +67,30 @@ package FetchStoreCounter {
     weaken($ref);
     undef $arr;
     is($ref, undef, 'ref is now undef after arr is cleared');
+
+    $ref = 'GH 19965';
+    {
+        local $@;
+        eval { weaken($ref); };
+        like($@, qr/^Can't weaken a nonreference/, "Can't weaken a defined nonreference");
+    }
+    {
+        local $@;
+        eval { unweaken($ref); };
+        like($@, qr/^Can't unweaken a nonreference/, "Can't unweaken a defined nonreference");
+    }
+
+    $ref = undef;
+    {
+        local $@;
+        eval { weaken($ref); };
+        ok(! $@, "Can weaken an undefined nonreference");
```

Comment:
You can call weaken on an undefined value, but it doesn't weaken it (that has no meaning).